src/config/constants.ts

```diff
@@ -22,7 +22,7 @@ export const EVENTS_POLLING_INTERVAL = 10 * 1000;
 
 export const HOST_ROLES = ['worker', 'master'];
 // Without undefined. Otherwise must conform generated Host['roles'] - see api/types.ts
-export type HostRolesType = 'master' | 'worker';
+export type Role = 'master' | 'worker' | undefined;
```

Comment:
There was reason for introducing this type without `undefined`. Otherwise let's use directly `Host['roles']`